### PR TITLE
Add Dockerfile + upgrade composer + upgrade php to 7.4 + fix

### PR DESCRIPTION
**Pull request by tomglvng**

### CODE REVIEW

**Review by AymericPlanche — COMMENTED** [review]
You need to update the [.circleci/config.yml](https://github.com/akeneo/api-php-client/blob/master/.circleci/config.yml#L6) and build the image to run the tests

**Review by LevFlavien — CHANGES_REQUESTED** [review]
Shouldn't CI use the same Dockerfile as well?

**Comment by tomglvng** [comment]
> You need to update the [.circleci/config.yml](https://github.com/akeneo/api-php-client/blob/master/.circleci/config.yml#L6) and build the image to run the tests. No ? thinking

Oups! My bad!

**Review by AymericPlanche — APPROVED** [review]

**Review by LevFlavien — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,40 +2,60 @@ version: 2.1
...
-        docker:
-            -   image: akeneo/php:7.2
+        machine:
+            image: ubuntu-2004:202111-01
         steps:
-            -   checkout
-            -   run: composer install
+            - run:
+                name: Change rights on project dir
+                command: sudo chmod -R 777 ../project
```
[the remark this answers is not in the document]
Comment (tomglvng) [reply .circleci/config.yml:10]: Yes, I tried to change user group, without success :(

**Pull request merged**